segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/TruncateOperation.java

```diff
@@ -121,6 +124,12 @@ private void logEnd() {
         val elapsed = timer.getElapsed();
         SLTS_TRUNCATE_LATENCY.reportSuccessEvent(elapsed);
         SLTS_TRUNCATE_COUNT.inc();
+        if (segmentMetadata.isStorageSystemSegment()) {
+            SLTS_SYSTEM_TRUNCATE_COUNT.inc();
+            val name = segmentMetadata.getName().substring(segmentMetadata.getName().lastIndexOf("/") + 1).replace("$", "_");
```

Comment:
Use the other overload of `replace` (The one taking `char` arguments). Its implementation is drastically simpler than the one taking a `String` - check it out.

Reply:
Fixed